File: Programmers/LEVEL_1/c++/문자열 내 마음대로 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
vector<string> strings;

vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;
    vector<pair<char, int> > temp;

    sort(strings.begin(), strings.end());   // 초반에 정렬을 하고 시작

    for (int i = 0; i < strings.size(); i++)
    {
        temp.push_back(make_pair(strings[i].at(n), i));
    }
    sort(temp.begin(), temp.end());

    for (int i = 0; i < temp.size(); i++)
    {
        answer.push_back(strings[temp[i].second]);
    }

    return answer;
}

File: Programmers/LEVEL_1/문자열 내 마음대로 정렬하기.js
function solution(strings, n) {
  var answer = [];

  strings.sort(function (a, b) {
    if (a[n] > b[n]) return 1;
    else if (b[n] > a[n]) return -1;
    else {
      if (a > b) return 1;
      else if (b > a) return -1;
      else return 0;
    }
  });

  answer = strings;

  return answer;
}
